subsys/bluetooth/host/conn.c

```diff
@@ -854,10 +854,18 @@ void bt_conn_data_ready(struct bt_conn *conn)
 		 *
 		 * This reference will be consumed when the conn is popped off
 		 * the list (in `get_conn_ready`).
+		 *
+		 * The `bt_dev.le.conn_ready` list is accessed by the tx_processor
+		 * which runs in a workqueue, but `bt_conn_data_ready` can be called
+		 * from different threads so we need to make sure that nothing will
+		 * trigger a thread switch while we are manipulating the list since
+		 * sys_slist_*() functions are not thread safe.
```

Comment:
`get_conn_ready` also manipulates the list by doing 

```
	if (should_stop_tx(conn)) {
		/* Move reference off the list and into the `conn` variable. */
		__maybe_unused sys_snode_t *s = sys_slist_get(&bt_dev.le.conn_ready);

		__ASSERT_NO_MSG(s == node);
		(void)atomic_set(&conn->_conn_ready_lock, 0);

		/* Append connection to list if it still has data */
		if (conn->has_data(conn)) {
			LOG_DBG("appending %p to back of TX queue", conn);
			bt_conn_data_ready(conn);
		}

		return conn;
	}
```

since `sys_slist_get` pops the list. Should this also be guarded by a `k_sched_lock`?

Reply:
That one is okay, because `get_conn_ready ` is called on `bt_conn_tx_processor` which is executed in the system workqueue. Since the system workqueue is being executed at a cooperative priority, we don't need it here.

Reply:
> we don't need it here.

_for now_ :D